packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemContextMenu.tsx

```diff
@@ -299,23 +294,47 @@ export const TreeItemContextMenu = ({
             <ListItemText primary="Delete" />
           </MenuItem>
         ) : null}
-        {state.menuEntries.map((entry) => (
-          <MenuItem
-            key={entry.id}
-            onClick={(_) => invokeContextMenuEntry(entry)}
-            data-testid={`context-menu-entry-${entry.label}`}
-            disabled={readOnly}
-            aria-disabled>
-            <ListItemIcon>
-              {entry.iconURL.length > 0 ? (
-                <IconOverlay iconURL={entry.iconURL} alt={entry.label} title={entry.label} />
-              ) : (
-                <div style={{ marginRight: '16px' }} />
-              )}
-            </ListItemIcon>
-            <ListItemText primary={entry.label} />
-          </MenuItem>
-        ))}
+        {state.menuEntries.map((entry) => {
+          const contributedTreeItemMenuContextComponents = treeItemBackendMenuContextComponents.filter(
+            (c) => c.identifier === entry.id
+          );
```

Comment:
This contract is too specific by requiring the id of the extension point to be the id of the menu item defined on the backend. With a data extension point, you can simply ask `canHandle` to the extension by giving it the menu entry. This way, consumers of this API could take over any menu entry they desire.

Reply:
fixed, there is now a `canHandle` the extension implements to tell the menu if it can render the entry. I also simplified the ids of the expand all tool to `expandAll`